python/src/com/jetbrains/python/psi/impl/references/PyQualifiedReference.java

```diff
@@ -158,16 +158,13 @@ private void addImplicitResolveResults(String referencedName, ResolveResultList
       }
     }
 
-    final Collection attributes = PyInstanceAttributeIndex.find(referencedName, project, scope);
-    for (Object attribute : attributes) {
-      if (!(attribute instanceof PyTargetExpression)) {
-        FileBasedIndex.getInstance().scheduleRebuild(StubUpdatingIndex.INDEX_ID,
-                                                     new Throwable(
-                                                       "found non-target expression object " + attribute + " in target expression list"));
-        break;
-      }
-      ret.add(new ImplicitResolveResult((PyTargetExpression)attribute, getImplicitResultRate((PyTargetExpression)attribute, imports)));
-    }
+    PyClassAttributesIndex.findClassAtrributes(referencedName, project, scope).forEach(classAttr -> {
```

Comment:
Let's drop `PyInstanceAttributeIndex` from the codebase since it's not used any longer.